NitroxClient/GameLogic/Crafting.cs

```diff
@@ -9,27 +12,453 @@ namespace NitroxClient.GameLogic
 {
     public class Crafting
     {
+        internal static Crafting Instance; 
+        private System.Collections.Generic.Dictionary<string, GameLogic.CraftingUsageInfo> fabricatorsInUseByRemotePlayers = new System.Collections.Generic.Dictionary<string, GameLogic.CraftingUsageInfo>();
+        private readonly System.Timers.Timer timer = new System.Timers.Timer(5000);
+        private readonly int remoteSignalTimeOut = 120; //in seconds. 2 Minutes, because if someone crafts multiple items, no new starttime is sent and we don't want to interrupt remote player.
+        private Constructable _lastHoveredConstructable = null;
         private readonly IPacketSender packetSender;
-
+        
         public Crafting(IPacketSender packetSender)
         {
+            Instance = this;
             this.packetSender = packetSender;
+            timer.AutoReset = true;
+            timer.Elapsed += new System.Timers.ElapsedEventHandler(OnTimerEvent);
+            timer.Start();
+        }
+
+        //Applies to Fabricator and Workbench
+        public void GhostCrafter_Post_OnHandHover(GameObject gameObject, GUIHand hand)
+        {
+#if TRACE && GAMEEVENTCRAFTING
+            NitroxModel.Logger.Log.Debug("GhostCrafter_Post_OnHandHover");
+#endif
+            //if Crafter is in use by remote Player > Display
+            string _crafterGuid = GuidHelper.GetGuid(gameObject);
+            bool _inUseByOther = false;
+            string _playerName = string.Empty;
+            lock (fabricatorsInUseByRemotePlayers)
```

Comment:
You don't need locks on the client side as unity always runs everything in a single thread included incoming packets.

Reply:
The lock is for the use of the System.Timer instead of a WindowsForms.Timer. I went to the System.Timer because WindowsForms is not part of the nitroxclient. System.Timer could be connected to a SynchronizingObject. But i don't know if there is one available which implements ISynchronizeInvoke from Subnautica and if using it would interfere with the GUI and Rendering logic from Subnautica. 
The System.Timer is relative easy to implement, but creates a WorkerThread. Thats the reason for the lock. 

An alternate approach would be, to not use the timer. Instead hook on the event when a remote player is disconnected. That also wouldn't need a "well defined" wait time for the duration until the usage is freed from the crafter. (which is always hard to define, because may be someone is reaching multiple storages from the crafter spot and can possibly use the crafter for a while, when crafting multiple items one after another).
In case something went wrong with unlocking the crafter after normal use, the player which is still   displayed as blocking user will then have to disconnect and reconnect, or use the crafter another time and free it again, which should be no problem for the players.